formatter/src/main/java/org/togetherjava/formatter/CodeSectionFormatter.java

```diff
@@ -0,0 +1,396 @@
+            appendNewLine();
+
+            if (type == TokenType.SEMICOLON) {
+                forLevel--;
+            }
+
+            applyIndentation = true;
+        }
+    }
+
+    /**
+     * Checks if a space should be put after a given token
+     *
+     * @param token token to check
+     * @return wether a space should be put after that token
+     * @author illuminator3
+     */
+    private boolean shouldPutSpaceAfter(Token token) {
+        TokenType type = token.type();
+
+        return isKeyword(token) || isOperator(token) || isParenthesisRule(token)
+                || type == TokenType.CLOSE_BRACKETS // put a space after e.g. 'try' or 'else' and
+                                                    // ] or if it's an operator
+                || type == TokenType.COMMA // for e.g. multiarg method calls or method parameters
+                || (type == TokenType.IDENTIFIER && !queue.isEmpty()
+                        && queue.peek().type() == TokenType.IDENTIFIER); // for double identifier
+                                                                         // in e.g. method
+                                                                         // declaration or enhanced
+                                                                         // for loops
+    }
+
+    /**
+     * Updates the enhanced for loop level
+     *
+     * @param type current token type
+     * @author illuminator3
+     */
+    private void updateEnhancedLevel(TokenType type) {
+        if (enhancedLevel != -1) {
+            switch (type) {
+                case OPEN_PARENTHESIS -> enhancedLevel++;
+                case CLOSE_PARENTHESIS -> enhancedLevel--;
+            }
+        }
+    }
+
+    /**
+     * Checks if a given token type belongs to a for loop (uses
+     * {@link SkippableLookaheadQueue#peek(int, Predicate)}
+     *
+     * @param type current token type
+     * @author illuminator3
+     */
+    private void checkFor(TokenType type) {
+        if (isIndexedForLoop(type)) { // if it's a for int loop then set the forLevel to 2
+            forLevel = 2;
+        } else if (isEnhancedForLoop(type)) {
+            enhancedLevel = 0;
+        }
+    }
+
+    /**
+     * Handles the case of being inside a generic type declaration
+     *
+     * @param token current token
+     * @author illuminator3
+     */
+    private void handleGeneric(Token token) {
+        TokenType type = token.type();
+
+        if (type == TokenType.EXTENDS || type == TokenType.SUPER) {
+            result.append(' ');
+        }
+
+        result.append(token.content());
+
+        if (type == TokenType.COMMA || type == TokenType.WILDCARD || type == TokenType.EXTENDS
+                || type == TokenType.SUPER) {
+            result.append(' ');
+        }
+
+        // ------------------------------------------------
+
+        switch (type) {
+            case SMALLER -> genericDepth++;
+            case BIGGER -> genericDepth--;
```

Comment:
Same as below, needs a default case. Because this will probably happen more, won't repeat it every time, you just check every switch.

Also took me some time to figure out what `SMALLER` means, it should be [`LESS_THAN`](https://en.wikipedia.org/wiki/Less-than_sign)

Reply:
(same as above)